### PR TITLE
GUI-based HuskTowns

### PR DESCRIPTION
**Pull request by Emibergo02**
not finished. 
I have to figure out how to put together the town list with the GUI since it isn't a OverviewCommand.
A friend of mine would beautify the GUIs when he will be available.
GUIs are fully customizable via JSON.
When I finish I'll make a User cache too as mentioned on Discord

### CODE REVIEW

File: bukkit/src/main/resources/plugin.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ libraries:
   - 'com.mysql:mysql-connector-j:${mysql_driver_version}'
   - 'org.mariadb.jdbc:mariadb-java-client:${mariadb_driver_version}'
   - 'org.xerial:sqlite-jdbc:${sqlite_driver_version}'
+  - 'io.netty:netty-all:${netty_version}'
```
Comment (WiIIiam278) [inline bukkit/src/main/resources/plugin.yml:13]: we shouldn't need to include this, right? Like pretty sure the Minecraft server bundles this

File: bukkit/src/main/java/net/william278/husktowns/BukkitHuskTowns.java [hunk 2]
```diff
@@ -297,6 +304,12 @@ public Manager getManager() {
         return manager;
     }
 
+    @NotNull
+    @Override
+    public GUIManager getGUIManager() {
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/BukkitHuskTowns.java:309]: Could we make "GUI" consistent between all class names? I.e "Gui" instead of "GUI"

File: bukkit/build.gradle [hunk 3]
```diff
@@ -19,6 +23,7 @@ dependencies {
     compileOnly 'com.github.MilkBowl:VaultAPI:1.7.1'
     compileOnly 'me.clip:placeholderapi:2.11.4'
     compileOnly 'net.luckperms:api:5.4'
+    compileOnly 'io.netty:netty-all:4.1.68.Final'
```
Comment (WiIIiam278) [inline bukkit/build.gradle:26]: Hmm.. where are we using this?

File: bukkit/build.gradle [hunk 4]
```diff
@@ -7,6 +7,10 @@ dependencies {
     implementation 'space.arim.morepaperlib:morepaperlib:0.4.3'
     implementation 'me.lucko:commodore:2.2'
     implementation 'net.william278:AdvancementAPI:master-SNAPSHOT'
+    implementation "xyz.xenondevs.invui:invui-core:1.15"
+    implementation "xyz.xenondevs.invui:inventory-access-r14:1.15"
+    implementation 'io.github.skytasul:glowingentities:1.3'
+    implementation 'com.github.Emibergo02:RedisChat:main-SNAPSHOT'
```
Comment (WiIIiam278) [inline bukkit/build.gradle:13]: not sure why we're including your plugin as an include dep - if there's functionality being added related to this it should be spun off into a separate PR imo

File: bukkit/build.gradle [hunk 5]
```diff
@@ -7,6 +7,10 @@ dependencies {
     implementation 'space.arim.morepaperlib:morepaperlib:0.4.3'
     implementation 'me.lucko:commodore:2.2'
     implementation 'net.william278:AdvancementAPI:master-SNAPSHOT'
+    implementation "xyz.xenondevs.invui:invui-core:1.15"
+    implementation "xyz.xenondevs.invui:inventory-access-r14:1.15"
+    implementation 'io.github.skytasul:glowingentities:1.3'
```
Comment (WiIIiam278) [inline bukkit/build.gradle:12]: Does this PR add glowing entity stuff as well?

File: bukkit/src/main/java/net/william278/husktowns/gui/deeds/TrustButton.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+    public TrustButton(DeedsGui deedsGui, BukkitHuskTowns plugin) {
+        this.deedsGui = deedsGui;
+        this.plugin = plugin;
+    }
+
+    @Override
+    public ItemProvider getItemProvider() {
+        if (deedsGui.selectedDeed.townClaim != null && deedsGui.selectedDeed.townClaim.claim().getType() == Claim.Type.PLOT) {
+            return new ItemBuilder(Material.PAPER).setDisplayName("§bTrust player");
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/gui/deeds/TrustButton.java:48]: style -- let's always use enclosed selection {}

File: bukkit/src/main/java/net/william278/husktowns/gui/deeds/DeedsGui.java [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+                        .toItemProvider(
+                                "%chunkx%", String.valueOf(townClaim.claim().getChunk().getX()),
+                                "%chunkz%", String.valueOf(townClaim.claim().getChunk().getZ())
+                        );
+            }
+
+            if (townClaim == null) {
+                return GuiSettings.getInstance().getDeedsGuiSettings().getItem("unclaimedItem").toItemProvider();
+            }
+            ItemBuilder ib;
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/gui/deeds/DeedsGui.java:173]: perhaps declare this as final and name it "builder"

File: bukkit/src/main/java/net/william278/husktowns/gui/deeds/DeedsGui.java [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+import xyz.xenondevs.invui.window.Window;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class DeedsGui extends PagedItemsGuiAbstract {
+    final OnlineUser onlineUser;
+    private final BukkitHuskTowns plugin;
+    DeedItem selectedDeed;
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/gui/deeds/DeedsGui.java:51]: protected or private

File: bukkit/src/main/java/net/william278/husktowns/gui/GuiSettings.java [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+import xyz.xenondevs.invui.item.ItemProvider;
+import xyz.xenondevs.invui.item.builder.ItemBuilder;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import static java.util.Map.entry;
+
+public class GuiSettings {
+    private static GuiSettings instance = new GuiSettings();
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/gui/GuiSettings.java:34]: I'd like to broadly avoid any static variable use - let's store this in the main GuiManager

File: bukkit/build.gradle (hunk 5 above)
Comment (Emibergo02) [reply bukkit/build.gradle:12]: nope srry didn't see that

File: bukkit/src/main/java/net/william278/husktowns/gui/GuiSettings.java [hunk 10]
```diff
@@ -117,15 +116,6 @@ public class GuiSettings {
                     entry("kickItem", new ItemDisplay(Material.BARRIER, 0, "Kick", new String[]{"Click to kick this member"}))
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/husktowns/gui/GuiSettings.java:116]: Map.entry - avoid static imports

**Comment by WiIIiam278** [comment]
Closing as stale for now :)
